### PR TITLE
uboot-lantiq: add support board ARV7519RW22-A-LT aka Livebox 2.1

### PR DESCRIPTION
**Pull request by Mandrake-Lee**
Package uboot-lantiq; added support for board ARV7519RW22-A-LT aka Livebox 2.1

The sources have been taken from https://github.com/danielschwierzeck/u-boot-lantiq with original author kept in the patch file.

Significant changes:

-Files and variables renamed in order to better match the hardware i.e arv7519.h ->arv7519rw22.h
-Variable CONFIG_SYS_DRAM_PROBE has been commented in file "arv7519rw22.h"; otherwise only half of the memory of the board will be detected.
-Variable CONFIG_SYS_BOOTM_LEN has been redefined to 16Mb. This was leading to crash with the latest initramfs images of Openwrt, which is a real inconvenient during development
-Defined kernel_addr in environment

The resulting binaries, both .bin an .asc (uart boot) have been tested with success.

Note for future development according @howl & @mkresin:
Recent versions could go beyond the reserved partition size for uboot, making uboot to break itself by writing env variables over itself. Two solutions could be:
A)Make bigger uboot partition also in openwrt image generation
B)Leave the same size but use compression for the phy firmwares.

Signed-off-by: Jorge Amorós-Argos <joramar76@yahoo.es>

Package uboot-lantiq; added support for board ARV7519RW22-A-LT aka Livebox 2.1

The sources have been taken from https://github.com/danielschwierzeck/u-boot-lantiq wi$

Significant changes:

-Files and variables renamed in order to better match the hardware i.e arv7519.h ->arv$
-Variable CONFIG_SYS_DRAM_PROBE has been commented in file "arv7519rw22.h"; otherwise $
-Variable CONFIG_SYS_BOOTM_LEN has been redefined to 16Mb. This was leading to crash w$
-Defined kernel_addr in environment

The resulting binaries, both .bin an .asc (uart boot) have been tested with success.

This is a (very late) amendment of https://github.com/openwrt/openwrt/pull/1883

### CODE REVIEW

**Comment by adrianschmutzler** [comment]
So, this is essentially a re-paste of the old PR: #1883

Note that you have partially duplicated your commit message.

**Comment by Mandrake-Lee** [comment]
> 
> 
> So, this is essentially a re-paste of the old PR: #1883
> 
> Note that you have partially duplicated your commit message.

Absolutely. I didn't know how to re-boot the old thread so I made a new pull request with your comments. Hope now everything is ok.

**Comment by adrianschmutzler** [comment]
Again, nobody even commented for three months after resubmitting this, and you didn't even update the commit message based on my feedback.

So, it seems like nobody is interested, and I'm closing this now. That's not your fault, but better destroy your remaining hopes now than wait for another 6 months with the same result.

**Comment by alimoche** [comment]
I have some feedback for this patch: I could not find any u-boot binaries for this board in the official builds,
see e.g. [19.07.7](https://downloads.openwrt.org/releases/19.07.7/targets/lantiq/xrx200/) there is, however, one custom u-boot in the device page: [arv7519rw22](https://openwrt.org/toh/arcadyan/arv7519rw22) that might have been built with a similar patch, so I gave this patch a try and rebuilt the u-boot for this board and it is working very well so far, I hope it helps

**Comment by howl** [comment]
I think this should be considered import because recently images for this devices are not being generated because is needed to tweak uboot. I also have the source code for the uboot commonly used for this device, the original author is @mkresin.

This pull request have some mistakes compared with the Mathias code.

**Comment by alimoche** [comment]
> This pull request have some mistakes compared with the Mathias code.

what are the mistakes? they may be easily fixed...

**Comment by howl** [comment]
Lack of brn boot and not sure if this code brings up all the memory or just half the ram present in the device, must be checked.

**Comment by alimoche** [comment]
In my case, u-boot shows the following message:

```
U-Boot 2013.10-openwrt4 (Apr 24 2021 - 23:09:33) arv7519rw22

Board: Arcadyan ARV7519RW22-A-LT
SoC:   Lantiq VRX288 v1.2
CPU:   500 MHz
IO:    250 MHz
BUS:   250 MHz
BOOT:  NOR
DRAM:  128 MiB
Flash: 32 MiB
In:    serial
Out:   serial
Err:   serial
Net:   ltq-eth
Hit any key to stop autoboot:  0 
``` 

and the device has this IC: EM68C16CWQD-25H

**Comment by Mandrake-Lee** [comment]
Hi,

This U-Boot is detecting all the RAM. This is the main difference I introduced against the code of @mkresin.

This is described in the body of the commit. See the top.

I believe this pull request was unmerged because I forgot to fulfil some minor change request of style or wording.  I didn't know the procedure so the time went up and I didn't have any more interest as this router is almost 10 years old.

But I assure it is fully functional. I was doing some development (network loading images via tftp) and I had no issue but to increase CONFIG_SYS_BOOTM_LEN to accomodate size of latest Openwrt images.

Regards,

**Comment by howl** [comment]
@Mandrake-Lee sorry for not re-reading the commit message, I wish @adschm reopens this pr. Try to add the code for brnboot, as it will be more complete.

I think this should be taken into account because the mentioned issue about the CONFIG_SYS_BOOTM_LEN and to not rely in external binaries.

**Comment by Mandrake-Lee** [comment]
> 
> 
> @Mandrake-Lee sorry for not re-reading the commit message, I wish @adschm reopens this pr. Try to add the code for brnboot, as it will be more complete.
> 
> I think this should be taken into account because the mentioned issue about the CONFIG_SYS_BOOTM_LEN and to not rely in external binaries.

Hi,

What do you mean by "brnboot"? This is the first (tiny) bootloader to be called when VR9 SoC powers up and then jumps to U-Boot.

As far as I know, there's no trace of this primer bootlader in the UART console so it's transparent to us.

Cheers

**Comment by howl** [comment]
It's possible to boot uboot from the stock brnboot. If I can, or perhaps @alimoche have it in it's code, we can bring that support altogether in the pr. The uboots than can be generated for this devices are, uboot to be write in the nor, uboot to be booted from the stock brnboot and the uboot to be booted from uart.

**Comment by Mandrake-Lee** [comment]
> 
> 
> It's possible to boot uboot from the stock brnboot. If I can, or perhaps @alimoche have it in it's code, we can bring that support altogether in the pr. The uboots than can be generated for this devices are, uboot to be write in the nor, uboot to be booted from the stock brnboot and the uboot to be booted from uart.

Hello,

I don't think there exists such a thing.

In the router instructions [Livebox2.1](https://openwrt.org/toh/arcadyan/arv7519rw22), the freshly made uboot is flashed into the partition 0, which is the first one.

This is what the esoteric commands 'U', '0', 'Y' does. 
```
[VR9 Boot]:u
	UPLOAD Flash
	---------------------------------------
	    Area            Address      Length
	---------------------------------------
	[0] Boot            0xB0000000     512K
	[1] Image 0         0xB1280000    6144K
	[2] Image 1         0xB1880000    6144K
	[3] Configuration   0xB1E80000    1024K
	[4] Boot Parameters 0xB1F80000     256K
	[5] Flash Image     0xB0000000   32768K
	[6] Linux Image     0xB0080000   24576K
	[7] Emergency Value 0xB1FC0000   2883584K
	---------------------------------------
	Enter area to UPLOAD:
```
The brnboot is stored into the SoC itself. So I guess that its purpose is to provide some minimal tools and then jump to the beginning of the NAND flash where the next bootloader will be waiting.

In other words, there's no especial uboot version baked against brnboot, at least as far (and little) I know.

**Comment by howl** [comment]
Yes it's possible to boot from brnboot and that commands are not esoteric. That we didn't documented it where the development started doesn't mean it exists.

I will try to focus on bringing all the necessary changes to give this device in pair with other similar arcadyan boards through your pr. But there is a need to attract attention from the adequate persons to merge this. I can't understand why this haven't take all the attention it needs because modified uboot is a necessary piece for this device to use openwrt.

**Comment by adschm** [comment]
> @Mandrake-Lee sorry for not re-reading the commit message, I wish @adschm reopens this pr.

Why would I? This has not worked out twice and the author stated that he isn't too interested either, so it's probably stuck a third time then.

**Comment by howl** [comment]
@adschm actually this device is source only because the need of a new uboot other than the precompiled ones found even in the wiki page. @Mandrake-Lee seems interested, the actual commit state seems fully functional just the need to increase CONFIG_SYS_BOOTM_LEN for the actual sizes.

Let's leave uboot as is this commit adding the CONFIG_SYS_BOOTM_LEN and then in future pr changes to bring brnboot. At least this device could be usable again with the actual 21.02 and snapshots.
